core/sql/optimizer/NAFileSet.cpp

```diff
@@ -161,6 +162,18 @@ Lng32 NAFileSet::getKeyLength()
 	return keyLength_;
 }
 
+// returns the length of the encoded key in bytes for this index
+Lng32 NAFileSet::getEncodedKeyLength()
+{
+	if(encodedKeyLength_ >0) return encodedKeyLength_;
+
+	for(CollIndex i=0;i<indexKeyColumns_.entries();i++)
+	{
+		encodedKeyLength_ += indexKeyColumns_[i]->getType()->getNominalSize();
```

Comment:
Hi,

I just submitted the rework.  Please review at your earliest convenience.
Thanks a lot!

On Tue, Oct 20, 2015 at 2:50 PM, Qifan Chen <qifan.chen@esgyn.com> wrote:

> Thanks!
>
> On Tue, Oct 20, 2015 at 2:02 PM, Hans Zeller <notifications@github.com>
> wrote:
>
>> In core/sql/optimizer/NAFileSet.cpp
>> <https://github.com/apache/incubator-trafodion/pull/132#discussion_r42538971>
>> :
>>
>> > @@ -161,6 +162,18 @@ Lng32 NAFileSet::getKeyLength()
>> >  	return keyLength_;
>> >  }
>> >
>> > +// returns the length of the encoded key in bytes for this index
>> > +Lng32 NAFileSet::getEncodedKeyLength()
>> > +{
>> > +	if(encodedKeyLength_ >0) return encodedKeyLength_;
>> > +
>> > +	for(CollIndex i=0;i<indexKeyColumns_.entries();i++)
>> > +	{
>> > +		encodedKeyLength_ += indexKeyColumns_[i]->getType()->getNominalSize();
>>
>> What's the reason for not calling the method that already does this,
>> NAType::getEncodedKeyLength()?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/apache/incubator-trafodion/pull/132/files#r42538971>.
>>
>
>
>
> --
> Regards, --Qifan
>
>


-- 
Regards, --Qifan